src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -349,6 +349,19 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 			{
 				DropTriggerEventExtendNames(dropStmt, schemaName, shardId);
 			}
+			else if (objectType == OBJECT_STATISTIC_EXT)
+			{
+				List *objectNameList = NULL;
+				foreach_ptr(objectNameList, dropStmt->objects)
+				{
+					RangeVar *stat = makeRangeVarFromNameList(objectNameList);
+
+					SetSchemaNameIfNotExist(&stat->schemaname, schemaName);
+
+					AppendShardIdToName(&stat->relname, shardId);
+					*objectNameList = *MakeNameListFromRangeVar(stat);
```

Comment:
This line is a little confusing. We're overwriting the List struct I guess? The List struct is quite different between PG12 and PG13, but maybe it works?

It would be nicer to do something like `shardStatisticsList = lappend(shardStatisticsList, MakeNameListFromRangeVar(stat));` and do `dropStmt->objects = shardStatisticsList` at the end.

Reply:
After resolving this it looks good to me.